java/clients/src/main/java/sleeper/clients/status/update/DeleteTable.java

```diff
@@ -0,0 +1,94 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package sleeper.clients.status.update;
+
+import com.amazonaws.services.dynamodbv2.AmazonDynamoDB;
+import com.amazonaws.services.dynamodbv2.AmazonDynamoDBClientBuilder;
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.AmazonS3ClientBuilder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import sleeper.clients.util.console.ConsoleInput;
+import sleeper.configuration.properties.instance.InstanceProperties;
+import sleeper.configuration.properties.table.S3TableProperties;
+import sleeper.configuration.properties.table.TableProperties;
+import sleeper.configuration.properties.table.TablePropertiesStore;
+import sleeper.core.table.TableIdentity;
+import sleeper.statestore.StateStoreProvider;
+
+import java.util.Optional;
+
+import static sleeper.clients.util.BucketUtils.deleteAllObjectsInBucketWithPrefix;
+import static sleeper.clients.util.ClientUtils.optionalArgument;
+import static sleeper.configuration.properties.instance.CdkDefinedInstanceProperty.DATA_BUCKET;
+import static sleeper.configuration.utils.AwsV1ClientHelper.buildAwsV1Client;
+import static sleeper.io.parquet.utils.HadoopConfigurationProvider.getConfigurationForClient;
+
+public class DeleteTable {
+    private static final Logger LOGGER = LoggerFactory.getLogger(DeleteTable.class);
+    private final AmazonS3 s3Client;
+    private final InstanceProperties instanceProperties;
+    private final TablePropertiesStore tablePropertiesStore;
+    private final StateStoreProvider stateStoreProvider;
+
+    public DeleteTable(AmazonS3 s3Client, AmazonDynamoDB dynamoDB, InstanceProperties instanceProperties) {
+        this(instanceProperties, s3Client, S3TableProperties.getStore(instanceProperties, s3Client, dynamoDB),
+                new StateStoreProvider(dynamoDB, instanceProperties, getConfigurationForClient()));
+    }
+
+    public DeleteTable(InstanceProperties instanceProperties, AmazonS3 s3Client, TablePropertiesStore tablePropertiesStore, StateStoreProvider stateStoreProvider) {
+        this.s3Client = s3Client;
+        this.instanceProperties = instanceProperties;
+        this.tablePropertiesStore = tablePropertiesStore;
+        this.stateStoreProvider = stateStoreProvider;
+    }
+
+    public void delete(String tableName) {
+        TableProperties tableProperties = tablePropertiesStore.loadByName(tableName);
+        tablePropertiesStore.deleteByName(tableName);
+        stateStoreProvider.getStateStore(tableProperties).clearSleeperTable();
+        TableIdentity tableId = tableProperties.getId();
+        deleteAllObjectsInBucketWithPrefix(s3Client, instanceProperties.get(DATA_BUCKET), tableId.getTableName());
```

Comment:
It doesn't seem like this is necessary, as there are no files stored under the table name?

Reply:
Thanks, removed